api/internal/controllers/analysis/analysis.go

```diff
@@ -147,11 +158,90 @@ func (c *Controller) hasDuplicatedHash(
 }
 
 func (c *Controller) publishInBroker(analysisID uuid.UUID) error {
-	response, err := c.GetAnalysis(analysisID)
+	res, err := c.GetAnalysis(analysisID)
 	if err != nil {
 		return err
 	}
 
 	return c.broker.Publish("", exchange.NewAnalysis,
-		exchange.Fanout, response.ToBytes())
+		exchange.Fanout, res.ToBytes())
+}
+
+//TODO:REMOVE ALL BELOW AFTER v2.10.0
+func (c *Controller) treatCompatibility(analysisEntity *analysis.Analysis) error {
+	if !c.existsDepecratedHashesSlice(analysisEntity.AnalysisVulnerabilities) {
+		return nil
+	}
+	hashSlice := make([]string, 0)
+
+	for i := range analysisEntity.AnalysisVulnerabilities {
+		hashSlice = append(hashSlice,
+			analysisEntity.AnalysisVulnerabilities[i].Vulnerability.DeprecatedHashes...)
+	}
+
+	if err := c.saveUpdates(hashSlice, analysisEntity); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (c *Controller) saveUpdates(hashSlice []string, analysisEntity *analysis.Analysis) error {
+	res := c.repoAnalysis.FindVulnerabilitiesByHashSliceInRepository(hashSlice, analysisEntity.RepositoryID)
+	if res.GetError() != nil {
+		return res.GetError()
+	}
+	mapHashToID, err := c.parseResIds(res)
+	if err != nil {
+		return err
+	}
+	query, values := c.mountUpdateQuery(analysisEntity, mapHashToID)
+
+	if err := c.repoAnalysis.RawQuery(query, values); err != nil {
+		return err
+	}
+	return nil
+}
+
+// mountUpdateQuery iterates over rawAnalysis.AnalysisVulnerabilities and
+// checks if some vuln.Vulnerability.DeprecatedHashes is present on
+// mapHashToId then creates and update statement to update the
+// deprecated Hash value to the new one (that is present in rawAnalysis.Vulnerability.VulnHash field)
+func (c *Controller) mountUpdateQuery(rawAnalysis *analysis.Analysis,
+	mapHashToID map[string]uuid.UUID) (string, []string) {
+	query := ""
+	values := make([]string, 0)
+	for i := range rawAnalysis.AnalysisVulnerabilities {
+		for _, hash := range rawAnalysis.AnalysisVulnerabilities[i].Vulnerability.DeprecatedHashes {
+			if mapHashToID[hash] != uuid.Nil {
+				query += "UPDATE vulnerabilities SET vuln_hash =?  where vulnerability_id = ? ;\n"
+				values = append(values, rawAnalysis.AnalysisVulnerabilities[i].Vulnerability.VulnHash, mapHashToID[hash].String())
+			}
+		}
+	}
+	return query, values
+}
+
+// existsDepecratedHashesSlice checks if []analysis.AnalysisVulnerabilities.Vulnerability
+// has a field called DeprecatedHashes
+func (c *Controller) existsDepecratedHashesSlice(slice []analysis.AnalysisVulnerabilities) bool {
```

Comment:
I don't think that `slice` is a good parameter name, maybe change to vulns? or vulnerabilities?